go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -158,6 +164,7 @@ type LocalnetConfig struct {
 	//    When set with 'Persistent', the assigned IP addresses will be persisted in `ipamclaims.k8s.cni.cncf.io` object.
 	// 	  Useful for VMs, IP address will be persistent after restarts and migrations. Supported when `ipam.mode` is `Enabled`.
 	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="layer2 topology config is immutable"
```

Comment:
for all optional, but immutable fields we need an extra validation on the `LocalnetConfig`, like so
`// +kubebuilder:validation:XValidation:rule="has(oldSelf.ipam) == has(self.ipam)", message="ipam is immutable"`
otherwise you can remove ipam completely, at least if it looks like
```
      ipam: 
        mode: Enabled
```
and it will be allowed. Or even better you can update
```
localnet:
  ipam:
    mode: Disabled
```
to
```
localnet:
  subnets: ["10.10.0.0/24"]
```
because ipam and subnets are optional fields :)